x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/request/azureaistudio/AzureAiStudioChatCompletionRequest.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference.external.request.azureaistudio;
+
+import org.apache.http.HttpHeaders;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.ByteArrayEntity;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.xcontent.XContentType;
+import org.elasticsearch.xpack.inference.external.request.HttpRequest;
+import org.elasticsearch.xpack.inference.external.request.Request;
+import org.elasticsearch.xpack.inference.services.azureaistudio.AzureAiStudioEndpointType;
+import org.elasticsearch.xpack.inference.services.azureaistudio.completion.AzureAiStudioChatCompletionModel;
+
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Objects;
+
+public class AzureAiStudioChatCompletionRequest extends AzureAiStudioRequest {
+    private final List<String> input;
+    private final AzureAiStudioChatCompletionModel completionModel;
+    private final boolean isRealtimeEndpoint;
+
+    public AzureAiStudioChatCompletionRequest(AzureAiStudioChatCompletionModel model, List<String> input) {
+        super(model);
+        this.input = Objects.requireNonNull(input);
+        this.completionModel = Objects.requireNonNull(model);
+        this.isRealtimeEndpoint = this.completionModel.endpointType() == AzureAiStudioEndpointType.REALTIME;
```

Comment:
I think we can get this field from `AzureAiStudioRequest` right?

Reply:
Ah - yep - forgot I added that into the superclass.